portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -39,8 +42,14 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String message = request.getParameter("message");
+
+    Entity commentEntity = new Entity("Comment");
+    commentEntity.setProperty("message", message);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
     if(!message.equals("")){
```

Comment:
message might be null. One easy solution would be  ```"".equals(message)``` to avoid NullPointerException. Or you might want to check message == null before storing it in database.

Reply:
If the message == null, should I throw a NullPointerException?

Reply:
For better fault tolerance, you could ignore the null message or return a 400 with  appropriate error message or other reasonable actions that won't cause an exception thrown.